plugins/linux-capture/pipewire.c

```diff
@@ -312,63 +312,63 @@ static inline bool has_effective_crop(obs_pipewire_data *obs_pw)
 		obs_pw->crop.height < obs_pw->format.info.raw.size.height);
 }
 
-static bool spa_pixel_format_to_drm_format(uint32_t spa_format,
-					   uint32_t *out_format)
-{
-	switch (spa_format) {
-	case SPA_VIDEO_FORMAT_RGBA:
-		*out_format = DRM_FORMAT_ABGR8888;
-		break;
-
-	case SPA_VIDEO_FORMAT_RGBx:
-		*out_format = DRM_FORMAT_XBGR8888;
-		break;
-
-	case SPA_VIDEO_FORMAT_BGRA:
-		*out_format = DRM_FORMAT_ARGB8888;
-		break;
-
-	case SPA_VIDEO_FORMAT_BGRx:
-		*out_format = DRM_FORMAT_XRGB8888;
-		break;
-
-	default:
-		return false;
-	}
+static const struct {
+	uint32_t spa_format;
+	uint32_t drm_format;
+	enum gs_color_format gs_format;
+	bool swap_red_blue;
+	const char *pretty_name;
+} supported_formats[] = {
```

Comment:
I like this. 
It probably won't solve the issue of having all formats in one place if we aim at adding webcam support via PipeWire. If we won't want to add YUV to RGBA convertion to the `graphics` API we have to use the `media` API with it's own set of supported formats. But that is sth. we can solve at that time.

Reply:
Yeah, I think YUV will be a different beast, we'll figure it out when we start adding these formats.